test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java

```diff
@@ -1754,89 +1929,79 @@ public void updatingBlankExternalIdFails() {
         ExternalIdentifier identifier = ExternalIdentifier.create(STUDY.getStudyIdentifier(), EXTERNAL_ID);
-        when(externalIdService.getExternalId(STUDY.getStudyIdentifier(), EXTERNAL_ID)).thenReturn(identifier);
+        when(externalIdService.getExternalId(STUDY.getStudyIdentifier(), EXTERNAL_ID, false)).thenReturn(identifier);
         
         participantService.updateParticipant(STUDY, PARTICIPANT);
         
         assertEquals(EXTERNAL_ID, account.getExternalId());
-        verify(externalIdService, never()).assignExternalId(STUDY, EXTERNAL_ID, HEALTH_CODE);
+        verify(externalIdService, never()).commitAssignExternalId(any());
     }
-    
-    @Test
-    public void removingManagedExternalIdWorks() {
-        mockHealthCodeAndAccountRetrieval();
-        STUDY.setExternalIdValidationEnabled(true);
-        ExternalIdentifier identifier = ExternalIdentifier.create(STUDY.getStudyIdentifier(), EXTERNAL_ID);
-        when(externalIdService.getExternalId(STUDY.getStudyIdentifier(), "newExternalId")).thenReturn(identifier);
-        
-        // This record has a different external ID than the mocked accound
-        StudyParticipant participant = new StudyParticipant.Builder().copyOf(PARTICIPANT)
-                .withExternalId(null).build();
-        participantService.updateParticipant(STUDY, participant);
-        
-        assertEquals(null, account.getExternalId());
-        verify(externalIdService).unassignExternalId(STUDY, EXTERNAL_ID, HEALTH_CODE);
-        verify(externalIdService).assignExternalId(STUDY, null, HEALTH_CODE);
-    }
-    
+
+    // Removed because you can no longer simply remove an external ID
+    // public void removingManagedExternalIdWorks();
+
     @Test
     public void createUnmanagedExternalIdWillAssign() {
         mockHealthCodeAndAccountRetrieval();
         STUDY.setExternalIdValidationEnabled(false);
-        when(externalIdService.getExternalId(STUDY.getStudyIdentifier(), EXTERNAL_ID)).thenReturn(null);
+        
+        when(externalIdService.getExternalId(STUDY.getStudyIdentifier(), EXTERNAL_ID, false)).thenReturn(null);
         
         participantService.createParticipant(STUDY, PARTICIPANT, false);
         
-        verify(externalIdService).assignExternalId(STUDY, EXTERNAL_ID, HEALTH_CODE);
+        verify(accountDao).createAccount(any(), accountCaptor.capture(), any());
+        assertEquals(EXTERNAL_ID, accountCaptor.getValue().getExternalId());
+        
+        verify(externalIdService).commitAssignExternalId(null);
     }
-    
+
     @Test
     public void addingUnmanagedExternalIdOnUpdateOK() {
         mockHealthCodeAndAccountRetrieval();
         STUDY.setExternalIdValidationEnabled(false);
         account.setExternalId(null);
         
         participantService.updateParticipant(STUDY, PARTICIPANT);
-        
+
         assertEquals(EXTERNAL_ID, account.getExternalId());
-        verify(externalIdService).assignExternalId(STUDY, EXTERNAL_ID, HEALTH_CODE);
```

Comment:
Should we verify commitAssignExternalId()? (Here and in changingUnmanagedExternalIdWorks().)